code/src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -112,43 +115,49 @@ public boolean getHasShotBall() {
         return m_hasShotBall;
     }
 
-    private boolean isUpToSpeed() {
-        if (m_shooterPID.atSetpoint() && m_shooterPID.getSetpoint() != 0) {
-            m_onTargetFor++;
-        } else {
-            m_onTargetFor = 0;
-        }
-
-        return m_onTargetFor >= pidOnTargetTicks;
+    private boolean hasBeamBroken() {
 
+        return m_beambreak.getRange() <= _config.Shooter.beamBrakeFinal;
     }
 
     public void periodic() {
-        double shooterSpeed = m_shooterPID.calculate(m_leftEncoder.getVelocity());
-        if (shooterSpeed > -0.2)
-            m_shooter.set(shooterSpeed);
-        else {
-            m_shooter.set(-0.2);
-        }
-        if (m_shooterPID.getSetpoint() == 0) {
+        /*
+         * double shooterSpeed = m_shooterPID.calculate(m_leftEncoder.getVelocity()); if
+         * (shooterSpeed > -0.2) m_shooter.set(shooterSpeed); else {
+         * m_shooter.set(-0.2); } if (m_shooterPID.getSetpoint() == 0) {
+         * m_shooter.set(0); }
+         */
+        if (m_isShooterOn) {
+            if (m_leftEncoder.getVelocity() > this.targetShooterRPM) {
+                m_shooter.set(_config.Shooter.lowerBangValue);
+            } else {
+
+                m_shooter.set(1);
+            }
+        } else {
             m_shooter.set(0);
         }
+        if (m_leftEncoder.getVelocity() > this.targetShooterRPM - _config.Shooter.pidTolerance) {
+            m_onTargetFor++;
+        } else {
+            m_onTargetFor = 0;
+        }
 
         // not having shot a ball implies that a button is being pressed, and we want
         // the feeder to be driven
         if (!this.m_hasShotBall) {
             if (this.m_feedBackward) {
-                this.m_feeder.set(-1);
+                this.m_feeder.set(-.5);
 
-            } else if (this.isUpToSpeed()) {
-                this.m_feeder.set(1);
+            } else {
+                this.m_feeder.set(.5);
                 this.m_isShooting = true;
             }
         }
 
-        // if we are shooting and the pid isn't at the setpoint, that means we've shot a
-        // ball
-        if (this.m_isShooting && !m_shooterPID.atSetpoint()) {
+        // if we are/were shooting and the pid isn't at the setpoint, that means we've
+        // shot a ball
+        if (this.m_isShooting && this.hasBeamBroken() && m_onTargetFor > 5) {
```

Comment:
move this to configs

Reply:
renamed to m_minOnTargetTicks

Justification: This is the minimum number of ticks that the limelight must be on target for it to be considered "ready to shoot".